keras_cv/models/object_detection/retinanet/retinanet_test.py

```diff
@@ -234,21 +234,81 @@ def test_saved_model(self, save_format, filename):
 
 @pytest.mark.large
 class RetinaNetSmokeTest(tf.test.TestCase):
-    def test_backbone_preset_weight_loading(self):
-        # Check that backbone preset weights loaded correctly
-        # TODO(lukewood): need to forward pass test once proper weights are
-        # implemented
-        keras_cv.models.RetinaNet.from_preset(
-            "resnet50_v2_imagenet",
-            num_classes=20,
-            bounding_box_format="xywh",
-        )
+    def test_backbone_preset(self):
+        weights = [
+            "csp_darknet_tiny",
+            "csp_darknet_s",
+            "csp_darknet_m",
+            "csp_darknet_l",
+            "csp_darknet_xl",
+            "efficientnetv2_s",
+            "efficientnetv2_m",
+            "efficientnetv2_l",
+            "efficientnetv2_b0",
+            "efficientnetv2_b1",
+            "efficientnetv2_b2",
+            "efficientnetv2_b3",
+            "mobilenet_v3_small",
+            "mobilenet_v3_small",
+            "resnet18",
+            "resnet34",
+            "resnet50",
+            "resnet101",
+            "resnet152",
+            "resnet18_v2",
+            "resnet34_v2",
+            "resnet50_v2",
+            "resnet101_v2",
+            "resnet152_v2",
+        ]
+        for weight in weights:
+            model = keras_cv.models.RetinaNet.from_preset(
+                weight,
+                num_classes=20,
+                bounding_box_format="xywh",
+            )
+            xs, _ = _create_bounding_box_dataset(bounding_box_format="xywh")
+            output = model(xs)
+            self.assertEqual(output["box"].shape, (xs.shape[0], 49104, 4))
 
     def test_full_preset_weight_loading(self):
-        # Check that backbone preset weights loaded correctly
-        # TODO(lukewood): need to forward pass test once proper weights are
-        # implemented
-        keras_cv.models.RetinaNet.from_preset(
+        model = keras_cv.models.RetinaNet.from_preset(
             "retinanet_resnet50_pascalvoc",
             bounding_box_format="xywh",
         )
+        xs, _ = _create_bounding_box_dataset(bounding_box_format="xywh")
+        output = model(xs)
+
+        expected_box = tf.constant(
+            [-0.40364307, 0.15742484, -1.3398054, 0.38104224]
+        )
+        self.assertAllClose(output["box"][0, 123, :], expected_box, atol=0.2)
```

Comment:
This tolerance seems way too high -- with a lower tolerance are you seeing failures?

Reply:
Yes!

Reply:
Ahh wait nevermind I see the problem 😅 

`_create_bounding_box_dataset` creates random noise, so we'd expect to see different results.

We should instead use `xs = tf.ones(1, 512, 512, 3)`